ZelBack/src/services/appsService.js

```diff
@@ -1859,23 +1859,30 @@ async function createAppVolume(appSpecifications, appName, isComponent, res) {
     }
 
     // if s flag create .stfolder
-    const containerDataFlags = appSpecifications.containerData.split('|')[0].split(':')[1] ? appSpecifications.containerData.split('|')[0].split(':')[0] : '';
-    if (containerDataFlags.includes('s')) {
-      const stFolderCreation = {
-        status: 'Creating .stfolder for syncthing...',
-      };
-      log.info(stFolderCreation);
-      if (res) {
-        res.write(serviceHelper.ensureString(stFolderCreation));
-      }
-      const execDIRst = `sudo mkdir -p ${appsFolder + appId}/.stfolder`;
-      await cmdAsync(execDIRst);
-      const stFolderCreation2 = {
-        status: '.stfolder created',
-      };
-      log.info(stFolderCreation2);
-      if (res) {
-        res.write(serviceHelper.ensureString(stFolderCreation2));
+    const containersData = appSpecifications.containerData.split('|');
+    // eslint-disable-next-line no-restricted-syntax
+    for (let i = 0; i < containersData.length; i += 1) {
+      const container = containersData[i];
+      const containerDataFlags = container.split(':')[1] ? container.split(':')[0] : '';
+      if (containerDataFlags.includes('s')) {
+        const containerFolder = i === 0 ? '' : `/appdata${container.split(':')[1].replace(containersData[0], '')}`;
```

Comment:
Why we did not need the /appdata before?
Are we sure that user input is correct? That it will be really followed by another /appdata/asd and not /appdataasd

Reply:
Also I don't understand the replacing. The containerData probably needs to come in some special way?
how about
/root|s:/root/syncthisfolder|s:/root/syncthisfoldertoo|0:/iamfirstocomponentdata
We should have some test suite and examples on what combinations are valid and what are not